pkg/genlib/generator_interface.go

```diff
@@ -1008,6 +1019,27 @@ func bindLongWithReturn(fieldCfg ConfigField, field Field, fieldMap map[string]a
 				dummyInt = fuzzyIntCounter(previous, fieldCfg.Fuzziness)
 			}
 
+			if fieldCfg.CounterReset != nil {
+				switch fieldCfg.CounterReset.Strategy {
+				case config.CounterResetStrategyRandom:
+					// 50% chance to reset
+					if customRand.Intn(2) == 0 {
+						dummyInt = 0
+					}
+				case config.CounterResetStrategyProbabilistic:
+					// Probability% chance to reset
+					if customRand.Intn(100) < int(*fieldCfg.CounterReset.Probability) {
+						dummyInt = 0
+					}
+				case config.CounterResetStrategyAfterN:
+					// Reset after N
+					if !state.counterReset && state.counter >= *fieldCfg.CounterReset.ResetAfterN {
+						dummyInt = 0
+						state.counterReset = true
+					}
```

Comment:
After resetting once, it is always set to reset. A bug?

```go
} else if state.counter < *fieldCfg.CounterReset.ResetAfterN {
	state.counterReset = false
}
```

Add this logic?

Reply:
Okay, then it is fine.

Reply:
Do you think it should be every N iterations or just once like it is now?

Reply:
IMO, every. If it is happening once, then why not every `n`.

Let's ask @aliabbas-elastic as well if he thinks that would be useful to have.